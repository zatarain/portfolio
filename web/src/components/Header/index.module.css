.header {
	min-height: 500px;
	color: var(--fg-secondary);
	position: relative;
	overflow: hidden;
}

.header::before {
	content: '';
	position: absolute;
	bottom: 10%;
	left: 50%;
	width: 120vh;
	height: 100vh;
	border-radius: 50%;
	transform-origin: bottom;
	transform: translateX(-50%) scale(3);
	background: var(--bg-primary);
	box-shadow: 0 5px 5px 0 var(--bg-primary);
	z-index: 1;
}

.header>* {
	z-index: 2;
	position: relative;
}

@media screen and (max-width: 850px) {
  .header {
    min-height: 30vh;
  }
}

@media screen and (max-width: 539px) {
  .header {
    min-height: 30vh;
  }
}
