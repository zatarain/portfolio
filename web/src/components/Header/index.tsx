import styles from './index.module.css'

interface Properties {
	data: {
		name: string
	}
}

const Header = ({ data }: Properties) => {
	return (
		<header className={styles.header}>
		</header>
	)
}

export default Header
