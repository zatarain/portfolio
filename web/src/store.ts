import { configureStore, ThunkAction, Action, createSlice } from '@reduxjs/toolkit'

export function makeStore() {
  const responsiveFlipper = createSlice({
    name: 'flipper',
    initialState: {
      responsive: false,
    },
    reducers: {
      flipResponsive: (state) => {
        state.responsive = !state.responsive
      }
    }
  })
  return configureStore({
    reducer: { flipper: responsiveFlipper.reducer },
  })
}

const store = makeStore()

export type AppState = ReturnType<typeof store.getState>

export type AppDispatch = typeof store.dispatch

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  AppState,
  unknown,
  Action<string>
>

export default store
