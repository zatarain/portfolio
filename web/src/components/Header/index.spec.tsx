import { Provider } from 'react-redux'
import store from '#store'
import { render } from '@testing-library/react'
import Header from '.'

describe('<Header />', () => {
	it('renders the component correctly with passed name', () => {

		const { asFragment, getByText } = render(
			<Provider store={store}>
				<Header data={{ name: 'My Name' }} />
			</Provider>
		)

		expect(getByText('My Name')).toBeInTheDocument()
		expect(asFragment()).toMatchSnapshot()
	})
})
