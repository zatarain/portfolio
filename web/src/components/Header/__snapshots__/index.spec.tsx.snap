// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the component correctly with passed name 1`] = `
<DocumentFragment>
  <header
    class="header"
  />
</DocumentFragment>
`;
